tests/IceRpc.Quic.Tests/Transports/QuicTransportSslAuthenticationTests.cs

```diff
@@ -0,0 +1,110 @@
+    {
+        if (!QuicConnection.IsSupported)
+        {
+            Assert.Ignore("Quic is not supported on this platform");
+        }
+    }
+
+    [Test]
+    public async Task Ssl_client_connection_connect_fails_when_server_provides_untrusted_certificate()
+    {
+        // Arrange
+        await using ServiceProvider provider = CreateServiceCollection()
+            .AddSingleton(
+                new SslServerAuthenticationOptions
+                {
+                    ServerCertificate = new X509Certificate2("server-untrusted.p12"),
+                })
+            .AddSingleton(
+                new SslClientAuthenticationOptions
+                {
+                    RemoteCertificateValidationCallback = (sender, certificate, chain, errors) => false
+                })
+            .BuildServiceProvider(validateScopes: true);
+
+        var sut = provider.GetRequiredService<ClientServerMultiplexedConnection>();
+
+        var listener = provider.GetRequiredService<IListener<IMultiplexedConnection>>();
+
+        // Start the TLS handshake.
+        Task clientConnectTask = sut.Client.ConnectAsync(default);
+
+        // Act/Assert
+        Assert.That(
+            async () => await listener.AcceptAsync(default),
+            Throws.TypeOf<AuthenticationException>());
+        Assert.That(async () => await clientConnectTask, Throws.TypeOf<AuthenticationException>());
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Security",
+        "CA5359:Do Not Disable Certificate Validation",
+        Justification = "The client doesn't need to validate the server certificate for this test")]
+    [Test]
+    public async Task Ssl_server_connection_connect_fails_when_client_provides_untrusted_certificate()
+    {
+        // Arrange
+        await using ServiceProvider provider = CreateServiceCollection()
+            .AddSingleton(
+                new SslServerAuthenticationOptions
+                {
+                    ClientCertificateRequired = true,
+                    RemoteCertificateValidationCallback = (sender, certificate, chain, errors) => false,
+                    ServerCertificate = new X509Certificate2("server.p12"),
+                })
+            .AddSingleton(
+                new SslClientAuthenticationOptions
+                {
+                    ClientCertificates = new X509CertificateCollection()
+                    {
+                        new X509Certificate2("client-untrusted.p12")
+                    },
+                    RemoteCertificateValidationCallback = (sender, certificate, chain, errors) => true
+                })
+            .BuildServiceProvider(validateScopes: true);
+
+        var sut = provider.GetRequiredService<ClientServerMultiplexedConnection>();
+        var listener = provider.GetRequiredService<IListener<IMultiplexedConnection>>();
+
+        // Start the TLS handshake.
+        var clientConnectTask = sut.Client.ConnectAsync(default);
```

Comment:
Sounds like this should be simply:
```csharp
await sut.Client.ConnectAsync(default);
```

Reply:
How would this complete before we call accept?

Reply:
From the previous version of this code, I thought the connectTask succeeded.